nixos/modules/services/networking/adguardhome.nix

```
@@ -0,0 +1,96 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.adguardhome;
+
+  args = concatStringsSep " " ([
+    "--no-check-update"
+    "--config ${cfg.confFile}"
+    "--work-dir ${cfg.workDir}"
+    "--host ${cfg.host}"
+    "--port ${toString cfg.port}"
+  ] ++ optional (cfg.logFile != null) "--logfile ${cfg.logFile}"
+  ++ optional (cfg.pidFile != null) "--pidfile ${cfg.pidFile}"
+  ++ optional cfg.verbose "--verbose");
+
+in
+{
+  options.services.adguardhome = with types; {
+    enable = mkEnableOption "AdGuard Home network-wide ad blocker";
+
+    confFile = mkOption {
+      default = "/etc/AdGuardHome.yaml";
+      type = path;
+      description = ''
+        Path to the config file.
+      '';
+    };
+
+    workDir = mkOption {
+      default = "/var/lib/AdGuardHome";
+      type = path;
+      description = ''
+        Path to the working directory.
+      '';
+    };
+
+    host = mkOption {
+      default = "0.0.0.0";
+      type = str;
+      description = ''
+        Host address to bind HTTP server on.
+      '';
+    };
+
+    port = mkOption {
+      default = 3000;
+      type = ints.u16;
+      description = ''
+        Port to serve HTTP pages on.
+      '';
+    };
+
+    logFile = mkOption {
+      default = "syslog";
+      type = nullOr (either path (enum [ "syslog" ]));
+      description = ''
+        Path to log file. If null: write to stdout; if 'syslog': write to system log.
+      '';
+    };
+
+    pidFile = mkOption {
+      default = "/run/AdGuardHome.pid";
+      type = nullOr path;
+      description = ''
+        Path to a file where PID is stored.
+      '';
+    };
+
+    verbose = mkOption {
+      default = false;
+      type = bool;
+      description = ''
+        Enable verbose output.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.adguardhome = {
+      description = "AdGuard Home: Network-level blocker";
+      after = [ "syslog.target" "network-online.target" ];
```

```suggestion
      after = [ "syslog.target" "network.target" ];
```

The program should graceful handle not connection.